vars/githubWorkflowRun.groovy

```diff
@@ -0,0 +1,115 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+/**
+
+  This step triggers workflow run on Github actions.
+
+  Example:
+     script {
+       def args = [
+           repo: "owner/repository",
+           workflow: "build.yml", 
+           ref: "main",
+           parameters: [
+             path: "filebeat"
+           ],
+           credentialsId: "github-workflow-token",
+           ghVersion: "2.1.0"]
+       def runId = githubWorkflowRun.triggerGithubActionsWorkflow(args)
+       def runInfo = githubWorkflowRun.getWorkflowRun(args + [runId: runId]) 
+   }
+*/
+
+import com.cloudbees.groovy.cps.NonCPS
+
+def call(Map args = [:]) {
+    // time in minutes to wait before build completed
+    def buildTimeLimit = args.get('buildTimeLimit', 30)
+    def runId = triggerGithubActionsWorkflow(args)
+    def startDate = new Date()
+    // the following loop intended to wait till triggered run completed
+    // or time out specifed by buildTimeLimit is reached
```

Comment:
pre-commit reported some code spells words

```suggestion
    // or time out specified by buildTimeLimit is reached
```

You can configure the `pre-commit` locally -> https://github.com/elastic/apm-pipeline-library/blob/master/docs/PRECOMMIT.md#installation

Reply:
fixed